src/libraries/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationElementExtension.cs

```diff
@@ -109,6 +118,8 @@ public string OuterName
 
         public string OuterNamespace
         {
+            [UnconditionalSuppressMessage("Trimming", "IL2026:RequiresUnreferencedCode", Justification = "Constructors not marked RequiresUnreferencedCode always set _outerName to a non-null value.")]
+            [UnconditionalSuppressMessage("AOT", "IL3050:RequiresDynamicCode", Justification = "Constructors not marked RequiresUnreferencedCode always set _outerName to a non-null value.")]
             get
             {
                 if (_outerName == null)
```

Comment:
Since this path is statically reachable, I suspect the linker will keep around bits of `XmlSerializer` and `DataContractSerializer`, even though it's never dynamically reachable from an app with no trim warnings. Is that right? Is there something simple we can do about it?

Reply:
You can split the `ExtensionDataWriter` nested class into a base abstract class (or interface) and a concrete implementation, and create the latter only in the trim-unsafe constructors.